internal/config/analytics.go

```diff
@@ -0,0 +1,61 @@
+package config
+
+import (
+	"errors"
+	"time"
+
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"github.com/spf13/viper"
+)
+
+// AnalyticsConfig defines the configuration for various mechanisms for
+// reporting and querying analytical data for Flipt.
+type AnalyticsConfig struct {
+	Enabled    bool             `json:"enabled,omitempty" mapstructure:"enabled" yaml:"enabled,omitempty"`
+	Clickhouse ClickhouseConfig `json:"clickhouse,omitempty" mapstructure:"clickhouse" yaml:"clickhouse,omitempty"`
```

Comment:
wdyt about nesting this under a subkey like `storage` or similar so that its not a 'top level' key

ie: `analytics.storage.clickhouse`

Reply:
@markphelps We could. I guess that semantically makes more sense.